File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.py
def solution(n):
    answer = 0
    answer = findPrime(n)
    return answer



def findPrime(num):
    count = 0
    num_list= [0] * (num+1)
    for i in range(2, num+1):
        num_list[i] = i
    for i in range(2, num+1):
        if(num_list[i] == 0):
            continue
        for j in range(2*i, num+1, i):
            num_list[j] = 0
    for i in range(2, num+1):
        if(num_list[i] != 0):
            count += 1
    return count
    
    

File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;


int solution(int n) {
    int answer = 0;
    vector<char> isPrime(n+1 , 1);
    isPrime[0] = 0;
    isPrime[1] = 0;
    for(int i = 2; i*i <= n; i++){
        if(isPrime[i]){
            for(int j = i*i; j <= n; j+=i){
                isPrime[j] = 0;
            }
        }
    }
    
    
    for(int a = 2 ; a <= n; a++){
        if(isPrime[a]){
            answer++;
        }
    }
    return answer;
}
